### PR TITLE
Bump thrift from 0.14.1 to 0.19.0

### PR DESCRIPTION
**Pull request by mmorel-35**
## Which problem is this PR solving?
- thrift 0.19.0 is available 

## Description of the changes
- update thrift from 0.14.1 to 0.19.0
- create a github workflow to lint and publish jaegertracing/thrift
- remove the previous Dockerfiles as they won’t be needed anymore
- QEMU steps are copied from docker-protobuf 

## How was this change tested?
- CI

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by mmorel-35** [comment]
cc @yurishkuro

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          type=raw,value=latest,enable=${{ github.ref == 'refs/heads/master' }}
+          type=semver,pattern={{version}}
+          type=ref,event=tag
+
+    - name: Login to DockerHub
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d  # v3.0.0
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (yurishkuro) [inline .github/workflows/build-and-publish.yml:49]: These were missing so the workflow failed. I just added them. Can you add `on: workflow_dispatch:` to the workflow triggers?

**Comment by yurishkuro** [comment]
@mmorel-35 just curious, do you actually need this? Are you still using Thrift-based Jaeger model?

**Comment by mmorel-35** [comment]
I don’t use it myself but as it is used in https://github.com/jaegertracing/jaeger/blob/main/Makefile.Thrift.mk

I thought it would be appropriate to keep it up to date

**Comment by yurishkuro** [comment]
More importantly is that it adds ARM64, I can actually run it on my laptop :-) The [build is running now](https://github.com/jaegertracing/docker-thrift/actions/runs/7413519812/job/20172618078), let's see if it succeeds and I will do a release.

**Comment by yurishkuro** [comment]
0.19 image published, trying it out in Jaeger https://github.com/jaegertracing/jaeger/pull/5080

Caching works great - the tagged release workflow ran in less than a minute (has to be on the same branch though).

**Pull request merged**